### PR TITLE
[review] Tokenizer control merging separators

### PR DESCRIPTION
**Pull request by dawagner**
Add a new optional behaviour for the Tokenizer, which makes it *not* ignore leading, trailing and consecutive delimiters.

Use it to fix a bug in domainGeneratorConnector that made it impossible to have empty values.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/334%23issuecomment-170871255%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/334%23issuecomment-170918347%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/334%23discussion_r49440000%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/334%23discussion_r49446245%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/334%23issuecomment-170920548%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/334%23issuecomment-170871255%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40krocard%20%40tcahuzax%20please%20review.%22%2C%20%22created_at%22%3A%20%222016-01-12T10%3A42%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222016-01-12T13%3A55%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%22%2C%20%22created_at%22%3A%20%222016-01-12T14%3A02%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2080b6d9d3754b632dd5bc06d1421c380d3756d372%20utility/Tokenizer.h%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/334%23discussion_r49440000%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22trailing%20and%20leading%22%2C%20%22created_at%22%3A%20%222016-01-12T11%3A01%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20Done%22%2C%20%22created_at%22%3A%20%222016-01-12T12%3A11%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/Tokenizer.h%3AL47-61%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/334#issuecomment-170871255'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> @krocard @tcahuzax please review.
- <a href='https://github.com/krocard'></a> :shipit:
- <a href='https://github.com/tcahuzax'></a> :-1:
- [x] <a href='#crh-comment-Pull 80b6d9d3754b632dd5bc06d1421c380d3756d372 utility/Tokenizer.h 15'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/334#discussion_r49440000'>File: utility/Tokenizer.h:L47-61</a></b>
- <a href='https://github.com/krocard'></a> trailing and leading
- <a href='https://github.com/dawagner'></a> :+1: Done

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/334?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/334?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/334?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/334'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/334'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by dawagner** [comment]
@krocard @tcahuzax please review.

File: utility/Tokenizer.h [hunk 1]
```diff
@@ -50,18 +47,14 @@ class Tokenizer : private utility::NonCopyable
      * @param[in] input The string to be tokenized
      * @param[in] delimiters A string containing all the token delimiters
      *            (hence, each delimiter can only be a single character)
+     * @param[in] mergeDelimiters If true, consecutive delimiters are considered
+     *            as one; trailing delimiters are also ignored.
```
Comment (krocard) [inline utility/Tokenizer.h]: trailing and leading

Comment (dawagner) [inline utility/Tokenizer.h]: :+1: Done

**Comment by krocard** [comment]
:shipit:

**Comment by tcahuzax** [comment]
:-1:

**Pull request merged**